packages/utils/src/dsn.ts

```diff
@@ -78,13 +81,20 @@ export class Dsn implements DsnComponents {
       }
     }
 
-    this._fromComponents({ host, pass, path, projectId, port, protocol: protocol as DsnProtocol, user });
+    this._fromComponents({ host, pass, path, projectId, port, protocol: protocol as DsnProtocol, publicKey });
   }
 
   /** Maps Dsn components into this instance. */
   private _fromComponents(components: DsnComponents): void {
+    if ('user' in components && !('publicKey' in components)) {
```

Comment:
tbh I'm not sure if we should bother with "that much" of a deprecation efforts. We could use JSDoc `@deprecated` everywhere only, as we will definitely leave both `user` and `publicKey` working in this major release anyway. We can reassign the data silently under the hood just like you do now.

Reply:
Okay, took out the log statement.